src/ahrs_driver.cpp

```diff
@@ -353,16 +363,42 @@ void ahrsBringup::processLoop()
         imu_data.linear_acceleration.y = imu_frame_.frame.data.data_pack.accelerometer_y;
         imu_data.linear_acceleration.z = imu_frame_.frame.data.data_pack.accelerometer_z;
       }
-      imu_data.orientation_covariance[0] = IMU_MAG_COV;
-      imu_data.orientation_covariance[4] = IMU_MAG_COV;
-      imu_data.orientation_covariance[8] = IMU_MAG_COV;
-      imu_data.angular_velocity_covariance[0] = IMU_GYRO_COV;
-      imu_data.angular_velocity_covariance[4] = IMU_GYRO_COV;
-      imu_data.angular_velocity_covariance[8] = IMU_GYRO_COV;
-      imu_data.linear_acceleration_covariance[0] = IMU_ACCEL_COV;
-      imu_data.linear_acceleration_covariance[4] = IMU_ACCEL_COV;
-      imu_data.linear_acceleration_covariance[8] = IMU_ACCEL_COV;
+      imu_data.orientation_covariance[0] = imu_mag_cov[0];
+      imu_data.orientation_covariance[4] = imu_mag_cov[1];
+      imu_data.orientation_covariance[8] = imu_mag_cov[2];
+      imu_data.angular_velocity_covariance[0] = imu_gyro_cov[0];
+      imu_data.angular_velocity_covariance[4] = imu_gyro_cov[1];
+      imu_data.angular_velocity_covariance[8] = imu_gyro_cov[2];
+      imu_data.linear_acceleration_covariance[0] = imu_accel_cov[0];
+      imu_data.linear_acceleration_covariance[4] = imu_accel_cov[1];
+      imu_data.linear_acceleration_covariance[8] = imu_accel_cov[2];
       imu_pub_.publish(imu_data);
+      // true East heading publish ----
+      tf::Quaternion q_new;
+      tf::Quaternion q_orig(
+        imu_data.orientation.x,
+        imu_data.orientation.y,
+        imu_data.orientation.z,
+        imu_data.orientation.w);
+      q_new = q_rot * q_orig;
+      q_new.normalize();
+
+      imu_trueEast_data.orientation.x = q_new.x();
+      imu_trueEast_data.orientation.y = q_new.y();
+      imu_trueEast_data.orientation.z = q_new.z();
+      imu_trueEast_data.orientation.w = q_new.w();
+      imu_trueEast_data.orientation_covariance[0] = imu_mag_cov[0];
+      imu_trueEast_data.orientation_covariance[4] = imu_mag_cov[1];
+      imu_trueEast_data.orientation_covariance[8] = imu_mag_cov[2];
+      imu_trueEast_data.angular_velocity_covariance[0] = imu_gyro_cov[0];
+      imu_trueEast_data.angular_velocity_covariance[4] = imu_gyro_cov[1];
+      imu_trueEast_data.angular_velocity_covariance[8] = imu_gyro_cov[2];
+      imu_trueEast_data.linear_acceleration_covariance[0] = imu_accel_cov[0];
+      imu_trueEast_data.linear_acceleration_covariance[4] = imu_accel_cov[1];
+      imu_trueEast_data.linear_acceleration_covariance[8] = imu_accel_cov[2];
```

Comment:
If `imu_trueEast_data` variable share the same covariance as `imu_data` variable, you can just write
```cpp
imu_trueEast_data.orientation_covariance = imu_data.orientation_covariance;
imu_trueEast_data.angular_velocity_covariance = imu_data.angular_velocity_covariance;
imu_trueEast_data.linear_acceleration_covariance = imu_data.linear_acceleration_covariance;
```

Reply:
Done